crates/verifier/src/stark/verify.rs

```diff
@@ -0,0 +1,171 @@
+extern crate alloc;
+
+use alloc::collections::BTreeMap;
+use alloc::fmt::Debug;
+use core::borrow::Borrow;
+use core::iter::repeat;
+use itertools::Itertools;
+
+use p3_field::{Field, FieldAlgebra};
+use p3_koala_bear::KoalaBear;
+use p3_symmetric::{CryptographicHasher, Permutation};
+use p3_util::reverse_slice_index_bits;
+use rayon::iter::IndexedParallelIterator;
+use rayon::iter::ParallelIterator;
+use rayon::slice::ParallelSlice;
+use serde::{Deserialize, Serialize};
+use zkm_core_executor::ZKMReduceProof;
+use zkm_core_machine::utils::log2_strict_usize;
+use zkm_recursion_core::air::{RecursionPublicValues, NUM_PV_ELMS_TO_HASH};
+use zkm_recursion_core::machine::RecursionAir;
+use zkm_stark::{
+    inner_perm, koala_bear_poseidon2::MyHash as InnerHash, CpuProver, MachineProof, MachineProver,
+    MachineVerificationError, StarkGenericConfig, DIGEST_SIZE,
+};
+
+use super::{HashableKey, InnerSC, ZKMVerifyingKey};
+
+const COMPRESS_DEGREE: usize = 3;
+pub type CompressAir<F> = RecursionAir<F, COMPRESS_DEGREE>;
+type CompressProver = CpuProver<InnerSC, CompressAir<<InnerSC as StarkGenericConfig>::Val>>;
+
+pub(crate) fn verify_stark_compressed_proof(
+    vk: &ZKMVerifyingKey,
+    proof: &ZKMReduceProof<InnerSC>,
+) -> Result<(), MachineVerificationError<InnerSC>> {
+    let allowed_vk_map: BTreeMap<[KoalaBear; DIGEST_SIZE], usize> =
+        bincode::deserialize(include_bytes!("../../../prover/dummy_vk_map.bin")).unwrap();
```

Comment:
It's better to use absolute path, and define it as a global variable,  like

const VK_MAP: &[u8] = include_bytes!(concat!(
    env!("CARGO_MANIFEST_DIR"), 
    "/prover/dummy_vk_map.bin"
));

We also need to make sure the other places that uses similar relative path to this way.  It's best to define a unified global variable for this.

Reply:
ok